nautobot/dcim/forms.py

```diff
@@ -1982,6 +2173,191 @@ class DeviceFilterForm(
     tags = TagFilterField(model)
 
 
+#
+# Modules
+#
+
+
+class ModuleForm(LocatableModelFormMixin, NautobotModelForm, TenancyForm):
+    manufacturer = DynamicModelChoiceField(
+        queryset=Manufacturer.objects.all(),
+        required=False,
+        initial_params={"module_types": "$module_type"},
+    )
+    module_type = DynamicModelChoiceField(
+        queryset=ModuleType.objects.all(),
+        query_params={"manufacturer": "$manufacturer"},
+    )
+    parent_module_bay_device_filter = DynamicModelChoiceField(
+        queryset=Device.objects.all(),
+        required=False,
+        label="Parent Device",
+        query_params={"has_empty_module_bays": True},
+        initial_params={"module_bays": "$parent_module_bay"},
+    )
+    parent_module_bay_device = DynamicModelChoiceField(
+        queryset=ModuleBay.objects.all(),
+        label="Parent Module Bay",
+        required=False,
+        query_params={"parent_device": "$parent_module_bay_device_filter", "has_installed_module": False},
+        initial_params={"pk": "$parent_module_bay", "parent_device__module_bays": "$parent_module_bay"},
+    )
+    parent_module_bay_module_filter = DynamicModelChoiceField(
+        queryset=Module.objects.all(),
+        required=False,
+        label="Parent Module",
+        query_params={"has_empty_module_bays": True},
+        initial_params={"module_bays": "$parent_module_bay"},
+    )
+    parent_module_bay_module = DynamicModelChoiceField(
+        queryset=ModuleBay.objects.all(),
+        label="Parent Module Bay",
+        required=False,
+        query_params={"parent_module": "$parent_module_bay_module_filter", "has_installed_module": False},
+        initial_params={"pk": "$parent_module_bay", "parent_module__module_bays": "$parent_module_bay"},
+    )
+    location = DynamicModelChoiceField(
+        queryset=Location.objects.all(),
+        required=False,
+        label="Location",
```

Comment:
Maybe some sort of help_text here as well like "Only specify Location if this module is not currently installed into a module bay"?

Reply:
Do you think that's necessary when the form is rendered only allowing one tab to be selected for device/module/location?

![image](https://github.com/nautobot/nautobot/assets/75227981/073932e2-d65b-4c6f-96da-65e260d67e12)

Reply:
Fair enough - probably not.